website/src/pages/Macro/Macro.jsx

```diff
@@ -0,0 +1,262 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import RingLoader from "react-spinners/RingLoader";
+import { Swiper, SwiperSlide } from "swiper/react";
+
+import '../Macro/macro.css'
+import "swiper/css";
+import "swiper/css/pagination";
+
+import { Mousewheel, Pagination, Keyboard } from "swiper";
+import Macrobench from '../../components/Macrobench/Macrobench';
+import MacrobenchMobile from '../../components/MacrobenchMobile/MacrobenchMobile';
+import { errorApi } from '../../utils/utils';
+
+const Macro = () => {
+    const urlParams = new URLSearchParams(window.location.search);
+// The following code sets up state variables `gitRefLeft` and `gitRefRight` using the `useState` hook.
+// The values of these variables are based on the query parameters extracted from the URL.
+
+// If the 'ltag' query parameter is null or undefined, set the initial value of `gitRefLeft` to 'Left',
+// otherwise, use the value of the 'ltag' query parameter.
```

Comment:
```suggestion
    // The following code sets up state variables `gitRefLeft` and `gitRefRight` using the `useState` hook.
    // The values of these variables are based on the query parameters extracted from the URL.

    // If the 'ltag' query parameter is null or undefined, set the initial value of `gitRefLeft` to 'Left',
    // otherwise, use the value of the 'ltag' query parameter.
```